jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsTerminal.kt

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.LangDataKeys
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.project.Project
+import icons.TerminalIcons
+import org.jetbrains.plugins.terminal.LocalTerminalDirectRunner
+import org.jetbrains.plugins.terminal.TerminalTabState
+import org.jetbrains.plugins.terminal.TerminalView
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.ConnectionSettings
+import software.aws.toolkits.jetbrains.core.credentials.safelyApplyTo
+import software.aws.toolkits.jetbrains.core.credentials.shortName
+import software.aws.toolkits.jetbrains.core.plugins.pluginIsInstalledAndEnabled
+import software.aws.toolkits.resources.message
+
+class StartAwsTerminal : AnAction(
+    message("aws.terminal.action"),
+    message("aws.terminal.action.tooltip"),
+    TerminalIcons.OpenTerminal_13x13
+) {
+    private val enabled by lazy {
+        pluginIsInstalledAndEnabled("org.jetbrains.plugins.terminal")
```

Comment:
Is terminal a dynamic plugin yet? If so, or if it ever will be, this needs to not be lazy. On the other hand, update is called a lot, and I'm not sure how cheap the plugin enabled check is.

Reply:
It is already dynamic.

Theres a message bus you can listen to to invalid the cache

Reply:
https://upsource.jetbrains.com/idea-ce/file/idea-ce-4682003011bb42ffdb872d081e79d300bb393d17/platform/platform-impl/src/com/intellij/ide/plugins/DynamicPlugins.kt

ApplicationLevel

Reply:
You actually dont need these checks at all I think...

It is already gated by the conditional plugin through the plgin.xml

Reply:
Ah, I see why, its not inside an ActionGroup but instead directly added to the title actions array, interesting...

Wonder if a action group works in the title actions.

Reply:
I think @rli found a problem where the plugin was installed but wasn't enabled - I wonder what happens in that case -- guess I can test it